### ISSUE TITLE
fix train so it works with validation set

### ISSUE THREAD

**Issue by NickleDave** [NickleDave/vak#57]

**Comment by NickleDave** [comment]
Was getting an error because `dataset.split` and the functions it uses, e.g. `splitalgos.bruteforce` assume there will always be a test set.

Fix this in a somewhat hacky way by re-writing all the functions to accept `test_dur` of `0`.
8de4cf503e6855198038377507e4fe5fddf6f325
df7116e5592ac0373b1dd5a0aebc371d797b04d4
f1ae0dd1a73a6c714bfa531d54a198a5f7a29f26

But now this muddles up `train_test_dur_split` function and a more general `split` function that isn't trying to have a `sklearn`-like API

**Comment by NickleDave** [comment]
We currently have this functionality so I am closing this issue
